### PR TITLE
Add int. tests for DataManagerFacility.move() method

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

Adds the integration tests for  the `DataManagerFacility.move()` method, see https://github.com/ome/omero-java-gateway/pull/3 .

# Testing this PR

Check that the tests make sense and pass.

# Related reading

https://github.com/ome/omero-java-gateway/pull/3

### CODE REVIEW

File: components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java [hunk 1]
```diff
@@ -354,6 +355,162 @@ public void testAttachAnnotation() throws Exception {
...
+
+        for (int i = 0; i < sources.size(); i++) {
+            DataObject src = sources.get(i);
+            DataObject tar = targets.get(i);
+            List<AnnotationData> attached = new ArrayList<AnnotationData>();
+            annos.forEach(anno -> {
+                try {
+                    attached.add(datamanagerFacility.attachAnnotation(rootCtx, anno, src));
+                } catch (DSOutOfServiceException | DSAccessException e) {
+                    e.printStackTrace();
```
Comment (jburel) [inline components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java]: you may want to fail if an error is thrown

File: components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java [hunk 2]
```diff
@@ -354,6 +355,162 @@ public void testAttachAnnotation() throws Exception {
...
+            datamanagerFacility.move(rootCtx, attached, src, tar);
+            List<AnnotationData> loaded = metadataFacility.getAnnotations(rootCtx, tar);
+            
+            Assert.assertEquals(attached.size(), loaded.size());
+            Set<Long> check = attached.stream().map(a -> a.getId()).collect(Collectors.toSet());
+            loaded.forEach(a -> Assert.assertTrue(check.contains(a.getId())));
+            
+            List<IObject> del = attached.stream().map(obj -> obj.asIObject()).collect(Collectors.toList());
+            datamanagerFacility.delete(rootCtx, del);
+            Thread.sleep(100);
```
Comment (jburel) [inline components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java]: why do you need it?

Comment (dominikl) [reply components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java]: As far as I remember I randomly hit some errors if I didn't wait there for the deletion to be finished. I'll test again without that line.

**Comment by joshmoore** [comment]
@dominikl : build is now green. Re-running west-ci should show you your tests passing.

**Comment by dominikl** [comment]
Thanks Josh. I have no idea what your `quick-gradle` does, but 👍

**Comment by joshmoore** [comment]
I cc'd you on https://github.com/ome/build-infra/pull/5: `quick-gradle` is just doing what was previously part of the Dockerfile directly:

 * checkout omero-build, merge if necessary
 * build and publishToMavenLocal
 * configure etc/omero.properties to use the SNAPSHOTS

File: components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java [hunk 3]
```diff
@@ -354,6 +355,161 @@ public void testAttachAnnotation() throws Exception {
...
+        annos.add(new TextualAnnotationData("test3"));
+        annos.add(new XMLAnnotationData("<test4/>"));
+
+        for (int i = 0; i < sources.size(); i++) {
+            DataObject src = sources.get(i);
+            DataObject tar = targets.get(i);
+            List<AnnotationData> attached = new ArrayList<AnnotationData>();
+            annos.forEach(anno -> {
+                try {
+                    attached.add(datamanagerFacility.attachAnnotation(rootCtx, anno, src));
```
Comment (jburel) [inline components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java:419]: improvement to gateway: add  ``attachAnnotations`` method

File: components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java [hunk 4]
```diff
@@ -354,6 +355,161 @@ public void testAttachAnnotation() throws Exception {
...
+                } catch (DSOutOfServiceException | DSAccessException e) {
+                    Assert.fail("Couldn't attach annotations.");
+                }
+            });
+            
+            datamanagerFacility.move(rootCtx, attached, src, tar);
+            List<AnnotationData> loaded = metadataFacility.getAnnotations(rootCtx, tar);
+            
+            Assert.assertEquals(attached.size(), loaded.size());
```
Comment (jburel) [inline components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java:429]: Could you add a check that the annotations have been moved from source
i.e. 
```
List<AnnotationData> loaded = metadataFacility.getAnnotations(rootCtx, src);

Assert.assertEquals(loaded.size(), 0);
```

File: components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java [hunk 5]
```diff
@@ -354,6 +355,161 @@ public void testAttachAnnotation() throws Exception {
...
+        DatasetData ds = new DatasetData();
+        ds.setName(UUID.randomUUID().toString());
+        ds = datamanagerFacility.createDataset(rootCtx, ds, proj1);
+
+        ProjectData proj2 = new ProjectData();
+        proj2.setName(UUID.randomUUID().toString());
+        proj2 = (ProjectData) datamanagerFacility.saveAndReturnObject(rootCtx,
+                proj2);
+
+        datamanagerFacility.move(rootCtx, Collections.singletonList(ds), proj1, proj2);
```
Comment (jburel) [inline components/tools/OmeroJava/test/integration/gateway/DataManagerFacilityTest.java]: good to check the move of more than one item

**Comment by joshmoore** [comment]
Note: implementation PR has been merged. I'll try to get this green and then merge.

**Comment by joshmoore** [comment]
Last attempt of quick-build is showing problems with gradle versions. I'm removing all of it, since there's now an RC that contains Dom's code. If this is green, I'd say let's merge and I can try again later.

**Comment by joshmoore** [comment]
Green. Merging for 5.5.0. Thanks, @dominikl !

**Pull request merged**